src/Inventory.ts

```diff
@@ -1,14 +1,35 @@
 import { Bot } from 'mineflayer'
-import { Callback } from './CollectBlock'
 import { Vec3 } from 'vec3'
-import { error } from './Util'
 import { Item } from 'prismarine-item'
-import { TemporarySubscriber } from './TemporarySubscriber'
-import { TaskQueue } from './TaskQueue'
-import { goals, ComputedPath } from 'mineflayer-pathfinder'
+import { goals } from 'mineflayer-pathfinder'
+import { NoChestsError } from './errors/NoChestsError'
+import { promisify } from 'util'
+import events from 'events'
 
 export type ItemFilter = (item: Item) => boolean
 
+export interface InventoryOptions {
+  bot: Bot
+  chestLocations: Vec3[]
+  itemFilter: ItemFilter
+}
+
+export async function emptyInventoryIfFull (options: InventoryOptions): Promise<void> {
+  if (options.bot.inventory.emptySlotCount() > 0) return
+  if (options.chestLocations.length === 0) throw new NoChestsError('There are no defined chest locations to empty inventory!')
+
+  // Shallow clone so we can safely remove chests from the list that are full.
+  const chestLocations = [...options.chestLocations]
```

Comment:
Minor: It might be worth noting that even though this says it's making a copy to allow mutations, it never removes chest locations it tries from the list. This was an issue in the existing code. I've made a [tracking issue](https://github.com/PrismarineJS/mineflayer-collectblock/issues/64). In my opinion this doesn't need to be addressed in this pr.

Reply:
The array is edited in the `getClosestChest()` function.